File: others/circular_queue.c
// Write a C program to implement the circular queue (enqueue, dequeue and display) using switch case and function
#include<stdio.h>

int high = -1, low = -1;
int arr[5];

void enqueue()
{
	if((high + 1) % 5 == low)
	{
		printf("Circular Queue Overflow\n");
	}
	else if(high == -1 && low == -1)
	{
		high = 0; low = 0;
		printf("Enter a number: ");
		scanf("%d",&arr[high]);
	}
	else
	{
		high = (high + 1) % 5;
		printf("Enter a number: ");
		scanf("%d",&arr[high]);
	}
}

void dequeue()
{
	if(low == -1 && high == -1)
	{
		printf("Circular Queue Underflow\n");
	}
	else if(low == high)
	{
		printf("Element deleted: %d\n",arr[low]);
		low = -1; high = -1;
	}
	else
	{
		printf("Element deleted: %d\n",arr[low]);
		low = (low + 1) % 5;
	}
}

void display()
{
	if(low == -1 && high == -1)
	{
		printf("Circular Queue Underflow\n");
	}
	else
	{
		int j;
		printf("Queue elements: ");
		for(j = low; j != high; j = (j+1) % 5)
		{
			printf("%d ",arr[j]);
		}
		printf("%d ",arr[j]);
		printf("\n");
	}
}

int main()
{
    int n = 1, a;
    while(n)
    {
        printf("\t Queue operation: - \n");
        printf("\t 1. Enqueue \n");
        printf("\t 2. Dequeue \n");
        printf("\t 3. Display \n");
        printf("\t 4. Exit \n");
        printf("Enter option: ");
        scanf("%d",&a);
        switch(a)
        {
            case 1:
                    enqueue();
                    break;
            case 2:
                    dequeue();
                    break;
            case 3:
                    display();
                    break;
            case 4:
                    n--;
                    printf("Operation ended\n");
                    break;
            default:
                    printf("You entered wrong choice\n");
        }
    }
}

File: others/CIRCULAR QUEUE.py
class Circular_Queue:
    circular_queue = []
    high, low = -1, -1
    def __init__(self):
        self.circular_queue_limit = 5	# circular queue size limit
    
    def enqueue(self):
        if len(Circular_Queue.circular_queue) == self.circular_queue_limit:
            print("CIRCULAR QUEUE Overflow\n")
        elif Circular_Queue.high == -1 and Circular_Queue.low == -1:
            Circular_Queue.high, Circular_Queue.low = 0, 0
            Circular_Queue.circular_queue.append(input("ENQUEUE ELEMENT: "))
            print('case 1')
        else:
            Circular_Queue.high = (Circular_Queue.high + 1) % self.circular_queue_limit
            Circular_Queue.circular_queue.append(input("ENQUEUE ELEMENT: "))
            print('case 2')
    
    def dequeue(self):
        if len(Circular_Queue.circular_queue) == 0:
            print("CIRCULAR QUEUE Underflow\n")
        elif Circular_Queue.high == Circular_Queue.low:
            print("POP ELEMENT:",Circular_Queue.circular_queue.pop(Circular_Queue.low))
            Circular_Queue.high, Circular_Queue.low = -1, -1
            print('case 3')
        else:
            print("DEQUEUE ELEMENT:",Circular_Queue.circular_queue.pop(0))
            #Circular_Queue.low = (Circular_Queue.low + 1) % self.circular_queue_limit
            print('case 4')
    
    def display(self):
        print("CIRCULAR QUEUE: ",Circular_Queue.circular_queue)

if __name__ == '__main__':
    while(True):
        print("\t Operation for CIRCULAR QUEUE: -")
        print("\t 	1. ENQUEUE")
        print("\t 	2. DEQUEUE")
        print("\t 	3. DISPLAY")
        print("\t 	4. EXIT")
        cq = Circular_Queue()
        option = int(input("Enter your option: "))
        if option == 1:
            cq.enqueue()
        elif option == 2:
            cq.dequeue()
        elif option == 3:
            print(cq.display())
        elif option == 4:
            break
